File: ios/NamazVakti/NamazVakti/Models/CalculationMethodRegistry.swift
import Foundation

struct CalculationMethodInfo: Identifiable, Hashable {
    let id: Int
    /// The issuing authority's own name. Institution names are proper nouns and stay
    /// untranslated; only the region descriptor below is localized.
    let name: String
    private let regionKey: String

    init(id: Int, name: String, regionKey: String) {
        self.id = id
        self.name = name
        self.regionKey = regionKey
    }

    var region: String { tr(regionKey) }
}

enum CalculationMethodRegistry {
    static let methods: [CalculationMethodInfo] = [
        CalculationMethodInfo(id: 13, name: "Türkiye (Diyanet)", regionKey: "region_tr_eu"),
        CalculationMethodInfo(id: 3, name: "Muslim World League (MWL)", regionKey: "region_global"),
        CalculationMethodInfo(id: 4, name: "Umm Al-Qura (Makkah)", regionKey: "region_gulf"),
        CalculationMethodInfo(id: 2, name: "ISNA", regionKey: "region_north_america"),
        CalculationMethodInfo(id: 15, name: "Moonsighting Committee", regionKey: "region_na_uk"),
        CalculationMethodInfo(id: 1, name: "Karachi (Univ. of Islamic Sciences)", regionKey: "region_south_asia"),
        CalculationMethodInfo(id: 5, name: "Egyptian General Authority", regionKey: "region_egypt"),
        CalculationMethodInfo(id: 16, name: "Dubai (Islamic Affairs)", regionKey: "region_uae"),
        CalculationMethodInfo(id: 11, name: "Singapore (MUIS)", regionKey: "region_sea"),
        CalculationMethodInfo(id: 9, name: "Kuwait", regionKey: "region_kuwait"),
        CalculationMethodInfo(id: 10, name: "Qatar", regionKey: "region_qatar")
    ]

    static func getMethodName(for id: Int) -> String {
        guard let method = methods.first(where: { $0.id == id }) else {
            return "Muslim World League (MWL)"
        }
        return "\(method.name) (\(method.region))"
    }
}

File: android/app/src/main/java/com/oktay/namaz/model/CalculationMethodRegistry.kt
package com.oktay.namaz.model

data class CalculationMethodInfo(
    val id: Int,
    val name: String,
    val region: String
)

object CalculationMethodRegistry {
    val methods = listOf(
        CalculationMethodInfo(13, "Türkiye (Diyanet)", "Türkiye & Avrupa"),
        CalculationMethodInfo(3, "Muslim World League (MWL)", "Europe & Global Default"),
        CalculationMethodInfo(4, "Umm Al-Qura (Makkah)", "Saudi Arabia & Gulf"),
        CalculationMethodInfo(2, "ISNA", "North America (USA & Canada)"),
        CalculationMethodInfo(15, "Moonsighting Committee", "North America & UK"),
        CalculationMethodInfo(1, "Karachi (Univ. of Islamic Sciences)", "Pakistan, India, Bangladesh"),
        CalculationMethodInfo(5, "Egyptian General Authority", "Egypt & North Africa"),
        CalculationMethodInfo(16, "Dubai (Islamic Affairs)", "United Arab Emirates"),
        CalculationMethodInfo(11, "Singapore (MUIS)", "Singapore & SE Asia"),
        CalculationMethodInfo(9, "Kuwait", "Kuwait"),
        CalculationMethodInfo(10, "Qatar", "Qatar")
    )

    fun getMethodName(id: Int): String {
        return methods.firstOrNull { it.id == id }?.let { "${it.name} (${it.region})" } ?: "Muslim World League (MWL)"
    }
}
